### ISSUE TITLE
panic: runtime error: index out of range

### ISSUE THREAD

**Issue by jstewart612** [improbable-eng/thanos#983]
**Thanos, Prometheus and Golang version used**
0.3.2 Docker image

**What happened**
Random PromQL queries querying data out of LTS cause the Thanos Store to spit out this error.

**What you expected to happen**
Not to get an error or return empty result.

**How to reproduce it (as minimally and precisely as possible)**:
Happening on any random query to LTS.

**Full logs to relevant components**
```
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | panic: runtime error: index out of range
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | goroutine 338336 [running]:
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | github.com/improbable-eng/thanos/pkg/objstore/azure.parseError(0xc42baa4480, 0x17b, 0x17b, 0x0)
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |   /go/src/github.com/improbable-eng/thanos/pkg/objstore/azure/helpers.go:68 +0x89
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | github.com/improbable-eng/thanos/pkg/objstore/azure.(*Bucket).IsObjNotFoundErr(0xc4204ae6e0, 0x11f6800, 0xc8c488a390, 0x0)
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |   /go/src/github.com/improbable-eng/thanos/pkg/objstore/azure/azure.go:147 +0x56
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | github.com/improbable-eng/thanos/pkg/objstore/azure.(*Bucket).Exists(0xc4204ae6e0, 0x1204d60, 0xc8dd565a00, 0xc894d5f8c0, 0x28, 0x0, 0xc42a7b1100, 0x112016b)
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |   /go/src/github.com/improbable-eng/thanos/pkg/objstore/azure/azure.go:220 +0x325
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | github.com/improbable-eng/thanos/pkg/objstore/azure.(*Bucket).getBlobReader(0xc4204ae6e0, 0x1204d60, 0xc8dd565a00, 0xc894d5f8c0, 0x28, 0x419670c, 0xab7fa, 0x120efa0, 0xc424418740, 0x0, ...)
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |   /go/src/github.com/improbable-eng/thanos/pkg/objstore/azure/azure.go:160 +0x2d2
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | github.com/improbable-eng/thanos/pkg/objstore/azure.(*Bucket).GetRange(0xc4204ae6e0, 0x1204d60, 0xc8dd565a00, 0xc894d5f8c0, 0x28, 0x419670c, 0xab7fa, 0x0, 0xc767a06980, 0xc767a069a8, ...)
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |   /go/src/github.com/improbable-eng/thanos/pkg/objstore/azure/azure.go:208 +0x77
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | github.com/improbable-eng/thanos/pkg/objstore.(*metricBucket).GetRange(0xc4201c3080, 0x1204d60, 0xc8dd565a00, 0xc894d5f8c0, 0x28, 0x419670c, 0xab7fa, 0x733efbd670c44669, 0x8174b71d1558f4ad, 0x449c68, ...)
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |   /go/src/github.com/improbable-eng/thanos/pkg/objstore/objstore.go:252 +0xf6
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | github.com/improbable-eng/thanos/pkg/store.(*bucketBlock).readChunkRange(0xc88c966bb0, 0x1204d60, 0xc8dd565a00, 0x0, 0x419670c, 0xab7fa, 0x0, 0x0, 0x0, 0x0, ...)
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |   /go/src/github.com/improbable-eng/thanos/pkg/store/bucket.go:1107 +0x16a
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | github.com/improbable-eng/thanos/pkg/store.(*bucketChunkReader).loadChunks(0xc7cdffb4a0, 0x1204d60, 0xc8dd565a00, 0xc87070dac0, 0xc, 0x10, 0x0, 0x4241f060419670c, 0x0, 0x0)
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |   /go/src/github.com/improbable-eng/thanos/pkg/store/bucket.go:1619 +0xc5
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | github.com/improbable-eng/thanos/pkg/store.(*bucketChunkReader).preload.func3(0x191ea687fe55213a, 0x9ea32b82d610bd3c)
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |   /go/src/github.com/improbable-eng/thanos/pkg/store/bucket.go:1605 +0x9f
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | github.com/improbable-eng/thanos/vendor/github.com/oklog/run.(*Group).Run.func1(0xc8857ac8a0, 0xc8857ac840, 0xc8a1bc3010)
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |   /go/src/github.com/improbable-eng/thanos/vendor/github.com/oklog/run/group.go:38 +0x27
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    | created by github.com/improbable-eng/thanos/vendor/github.com/oklog/run.(*Group).Run
thanos_store.1.1nsnmzm0migb@thanos-cluster001.tools.useast2.rentpath.com    |   /go/src/github.com/improbable-eng/thanos/vendor/github.com/oklog/run/group.go:37 +0xa8
```

**Anything else we need to know**

**Comment by bwplotka** [comment]
Cool, looks like Azure code issue, we will look, thanks!

**Comment by GiedriusS** [comment]
Seems like it has been fixed in master recently. Care to give it a try? (:

**Comment by jstewart612** [comment]
It will take me a few days to get to do so.  Is the next release already scheduled to include that commit?  Can you link it so I can track until I can get to trying this (we normally build atop the already pre-built docker image so it will take a bit of re-tooling to both build from source and put in-line)?

**Comment by jstewart612** [comment]
Okay, I've reloaded the new version.  No issues so far.  I'll burn it in over the next business day or two and come back with final word.

**Comment by povilasv** [comment]
feel free to close the issue if everything works out well.

**Comment by GiedriusS** [comment]
#961 talking about this one :P

**Comment by jstewart612** [comment]
Sadly, not working well... all my longer term queries do this now:

```
thanos_query.0.rremp8jg2q2x@thanos-cluster003.tools.useast2.rentpath.com    | {"caller":"proxy.go:346","err":"failed to receive any data from Addr: store:10903 Labels: [] Mint: 1552420800000 Maxt: 1553853600000: context canceled","level":"warn","msg":"returning partial response","ts":"2019-03-29T11:48:50.983610106Z"}
thanos_query.0.rremp8jg2q2x@thanos-cluster003.tools.useast2.rentpath.com    | {"caller":"proxy.go:346","err":"failed to receive any data from Addr: thanos-services001.ci.useast2.rentpath.com:10901 Labels: [{monitor thanos-services001.ci.useast2.rentpath.com {} [] 0} {region useast2 {} [] 0}] Mint: 1553760000000 Maxt: 9223372036854775807: context canceled","level":"warn","msg":"returning partial response","ts":"2019-03-29T11:48:50.983761806Z"}
thanos_query.0.rremp8jg2q2x@thanos-cluster003.tools.useast2.rentpath.com    | {"caller":"proxy.go:346","err":"failed to receive any data from Addr: thanos.vip.tools.scen1.rentpath.com:10901 Labels: [] Mint: 1550707200000 Maxt: 9223372036854775807: context canceled","level":"warn","msg":"returning partial response","ts":"2019-03-29T11:48:50.983806006Z"}
thanos_query.0.rremp8jg2q2x@thanos-cluster003.tools.useast2.rentpath.com    | {"caller":"proxy.go:346","err":"failed to receive any data from Addr: thanos-apps001.qa.useast2.rentpath.com:10901 Labels: [{monitor thanos-apps001.qa.useast2.rentpath.com {} [] 0} {region useast2 {} [] 0}] Mint: 1553760000000 Maxt: 9223372036854775807: context canceled","level":"warn","msg":"returning partial response","ts":"2019-03-29T11:48:50.983862706Z"}
thanos_query.0.rremp8jg2q2x@thanos-cluster003.tools.useast2.rentpath.com    | {"caller":"proxy.go:346","err":"failed to receive any data from Addr: thanos-services001.tools.useast2.rentpath.com:10901 Labels: [{monitor thanos-services001.tools.useast2.rentpath.com {} [] 0} {region useast2 {} [] 0}] Mint: 1553760000000 Maxt: 9223372036854775807: context canceled","level":"warn","msg":"returning partial response","ts":"2019-03-29T11:48:50.983904106Z"}
thanos_query.0.rremp8jg2q2x@thanos-cluster003.tools.useast2.rentpath.com    | {"caller":"proxy.go:346","err":"failed to receive any data from Addr: app-manager.vip.useast2.rentpath.com:10901 Labels: [{monitor app-manager.vip.useast2.rentpath.com {} [] 0} {region useast2 {} [] 0}] Mint: 1553760000000 Maxt: 9223372036854775807: context canceled","level":"warn","msg":"returning partial response","ts":"2019-03-29T11:48:50.984016906Z"}
thanos_query.0.rremp8jg2q2x@thanos-cluster003.tools.useast2.rentpath.com    | {"caller":"proxy.go:346","err":"failed to receive any data from Addr: thanos-services001.r.useast2.rentpath.com:10901 Labels: [{monitor thanos-services001.r.useast2.rentpath.com {} [] 0} {region useast2 {} [] 0}] Mint: 1553760000000 Maxt: 9223372036854775807: context canceled","level":"warn","msg":"returning partial response","ts":"2019-03-29T11:48:50.984058306Z"}
thanos_query.0.rremp8jg2q2x@thanos-cluster003.tools.useast2.rentpath.com    | {"caller":"proxy.go:346","err":"failed to receive any data from Addr: app-manager.vip.qa.useast2.rentpath.com:10901 Labels: [{monitor app-manager.vip.qa.useast2.rentpath.com {} [] 0} {region useast2 {} [] 0}] Mint: 1553760000000 Maxt: 9223372036854775807: context canceled","level":"warn","msg":"returning partial response","ts":"2019-03-29T11:48:50.984102206Z"}
thanos_query.0.rremp8jg2q2x@thanos-cluster003.tools.useast2.rentpath.com    | {"caller":"proxy.go:346","err":"failed to receive any data from Addr: thanos-services001.qa.useast2.rentpath.com:10901 Labels: [{monitor thanos-services001.qa.useast2.rentpath.com {} [] 0} {region useast2 {} [] 0}] Mint: 1553760000000 Maxt: 9223372036854775807: context canceled","level":"warn","msg":"returning partial response","ts":"2019-03-29T11:48:50.984149006Z"}
```

**Comment by jstewart612** [comment]
Issue does appear to have been fixed in #961 .  I thought it may not have been due to store connection errors I was getting post start, but I had jumped the gun posting logs before my sidecars had started.
